code-challenges.test.js

```diff
@@ -10,52 +10,124 @@
+  // Expected output: "L4ck4d41s1c4l"
+  const secretCodeWord2 = "Gobbledygook";
+  // Expected output: "G0bbl3dyg00k"
+  const secretCodeWord3 = "Eccentric";
+  // Expected output: "3cc3ntr1c"
+  it("takes in a string and returns a coded message, converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", () => {
+    expect(convertsCodedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l");
+    expect(convertsCodedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k");
+    expect(convertsCodedMessage(secretCodeWord3)).toEqual("3cc3ntr1c");
+  });
+});
+
+// RED: variable not defined
 
 // b) Create the function that makes the test pass.
 
+// create a function that takes in a string and returns a coded message
+//
+// define characters which will be swapped out and determine ALL will be replaced, not just the first value when iteration occurs (/[]/g)
+// use built in method .replace()
+// create an object with key value pairs
 
+const convertsCodedMessage = (string) => {
+  return string.replace(
+    /[aeioAEIO]/g,
+    (value) => ({ a: 4, A: 4, e: 3, E: 3, i: 1, I: 1, o: 0, O: 0 }[value])
+  );
+};
+
+// GREEN
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
-// Expected output: ["Apple", "Banana", "Orange"]
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
-
-
+describe("wordsThatInclude", () => {
+  const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"];
+  const letterA = "a";
+  // Expected output: ["Apple", "Banana", "Orange"]
+  const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"];
+  const letterE = "e";
+  // Expected output: ["Cherry", "Blueberry", "Peach"]
+  it("takes in an array of words and a single letter and returns all the words that contain that particular letter ", () => {
+    expect(wordsThatInclude(arrayOfWords1, letterA)).toEqual([
+      "Apple",
+      "Banana",
+      "Orange",
+    ]);
+    expect(wordsThatInclude(arrayOfWords2, letterE)).toEqual([
+      "Cherry",
+      "Blueberry",
+      "Peach",
+    ]);
+  });
+});
+
+// RED: variable not defined
 
 // b) Create the function that makes the test pass.
 
+// create a function that takes in an array of words and a single letter, returns all the words that contain that letter
+// iterate over array to identify words containing the letter
+// return word(s) containing the letter in array
+// use built-in method .toLowerCase() to include both cases of the letter taken in by the function
+
+const wordsThatInclude = (array, letter) => {
+  wordsInclArr = array.filter((value) => {
+    return value.toLowerCase().includes(letter);
+  });
+  return wordsInclArr;
+};
```

Comment:
Great! Just a small note that if you are creating a variable only to return it you can just return the logic. 
```
const wordsThatInclude = (array, letter) => {
  return array.filter((value) => {
    return value.toLowerCase().includes(letter);
  });
};
```